src/sparseml/exporters/transforms/utils/helpers.py

```diff
@@ -125,22 +126,17 @@ def assert_node_type(node: NodeProto, op: Union[List[str], Set[str], str]) -> bo
 def quantize_array(
     array: numpy.ndarray, scale: float, zero_point: int, dtype: Any = numpy.uint8
 ) -> numpy.ndarray:
+    try:
+        from sparseml.exporters.transforms.utils.helpers_torch import (
+            quantize_array_torch,
+        )
 
-    if dtype == numpy.uint8:
-        tensor_dtype = torch.quint8
-    elif dtype == numpy.int8:
-        tensor_dtype = torch.qint8
-    elif dtype == numpy.int32:
-        tensor_dtype = torch.qint32
-
-    tensor = torch.Tensor(array.copy()).to(torch.float32)
-    if isinstance(scale, numpy.ndarray):
-        scale = scale.item()
-    if isinstance(zero_point, numpy.ndarray):
-        zero_point = zero_point.item()
-
-    quant_tensor = torch.quantize_per_tensor(tensor, scale, zero_point, tensor_dtype)
-    return quant_tensor.int_repr().numpy()
+        return quantize_array_torch(array, scale, zero_point, dtype)
+    except ModuleNotFoundError as err:
+        _LOGGER.debug(f"Error: {err}. Defaulting to numpy implementation.")
+        dmin = numpy.iinfo(dtype).min
+        dmax = numpy.iinfo(dtype).max
+        return ((array / scale).round() + zero_point).clip(dmin, dmax).astype(dtype)
```

Comment:
Could we just always use this numpy version? I assume this is the same thing torch is doing?

Reply:
@bfineran: "I think it was some rounding error reason relative to the torch quantization execution for why we switched"

Reply:
I ran a few tests to check whether the two implementations match. They did, but I was probably not too thorough to spot the difference.

Reply:
"Fun" challege: try to find the pytorch implementation of this

Reply:
This is the best I could find about what they do. https://github.com/pytorch/pytorch/wiki/torch_quantization_design_proposal#quantized-representation

I couldn't find the C++ code where this actually happens 😩 

We could try running all our regression testing with this version?

Reply:
Okay I was able to find some of the cuda kernels at least https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cuda/FakeQuantizeCore.cu

Seems like they may do some weird rounding stuff. I suspect your implementation may different for values in a certain range?